src/lib/isIdentityCard.js

```diff
@@ -30,6 +30,31 @@ const validators = {
 
     return sanitized.endsWith(controlDigits[number % 23]);
   },
+  'he-IL': (str) => {
+    const DNI = /^\d+$/;
+
+    // sanitize user input
+    const sanitized = str.trim();
+
+    // validate the data structure
+    if (!DNI.test(sanitized)) {
+      return false;
+    }
+
+    const id = sanitized;
+
+    if (id.length !== 9 || isNaN(id)) {
```

Comment:
@profnandaa isn't this check unnecessary if we add the size limit in the regex?

Reply:
Sure, that's true. That was oversight from my end. You can do a quick update PR on this?

```
const DNI = /^\d{9}$/;
```

cc. @perimiter

Reply:
thats true, but you still need the if statement for the `isNaN()`

Reply:
@perimiter your regex is searching only for digits, the function will always exit before the call to ```isNaN``` if it's not a number

Reply:
your right, i opened a PR with the new changes (#1048)